src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt

```diff
@@ -290,12 +296,19 @@ open class RsDefaultInsertHandler : InsertHandler<LookupElement> {
             is RsTraitItem -> appendSemicolon(context, curUseItem)
             is RsStructItem -> appendSemicolon(context, curUseItem)
 
-            is RsFunction -> {
-                if (curUseItem != null) {
+            is RsFunction -> when {
+                curUseItem != null -> {
                     appendSemicolon(context, curUseItem)
-                } else {
+                }
+                element.isProcMacroDef -> {
+                    if (element.isBangProcMacroDef && !context.nextCharIs('!')) {
+                        document.insertString(context.selectionEndOffset, "!()")
```

Comment:
@dima74 BTW we use special heuristics to complete proper braces for macro calls (see `org.rust.lang.core.psi.ext.RsMacroDefinitionBase#getPreferredBraces` and `org.rust.lang.core.psi.ext.RsMacroDefinitionBaseKt#guessPreferredBraces`) based on documentation. It seems the same thing can be used for procedural bang macros as well